project/npda/management/commands/upload_csv.py

```diff
@@ -14,31 +26,165 @@ class Command(BaseCommand):
     )
 
     def add_arguments(self, parser):
-        parser.add_argument("--file", type=str, required=True, help="File to upload")
+        parser.add_argument(
+            "--file",
+            type=str,
+            required=True,
+            help="File to upload"
+        )
 
         parser.add_argument(
-            "--user", type=int, default=1, help="PK of the user uploading the file"
+            "--user",
+            type=int,
+            required=True,
+            help="NPDA ID (primary key) of the user uploading the file"
         )
 
         parser.add_argument(
             "--pz-code",
             type=str,
-            default="PZ999",
             help="PZ code of the PDU for the upload",
         )
 
+        parser.add_argument(
+            "--audit-year",
+            type=str,
+            required=True,
+            help="Audit year for the upload",
+        )
+
+        parser.add_argument(
+            "--import-as-questionnaire-entries",
+            action="store_true",
+            help="""Import data from the file as if it were submitted using the questionnaire.
+                    Assign each row to a PDU using the 'PDU Number' column".
+                    Will create a questionnaire submission for each PDU in the file.
+                    If any submissions already exist no data will be imported.
+                    Designed for bulk import patient data from the old platform."""
+        )
+
+    def upload_csv_to_single_pdu(self, audit_year, pdu_pz_code, user, parsed_csv, csv_file_bytes, csv_file_name):
+        pdu = PaediatricDiabetesUnit.objects.get(pz_code=pdu_pz_code)
+
+        submission = async_to_sync(create_csv_submission)(
+            pdu=pdu,
+            audit_year=audit_year,
+            csv_file_bytes=csv_file_bytes,
+            csv_file_name=csv_file_name,
+            user=user
+        )   
+    
+        errors = async_to_sync(csv_upload)(
+            dataframe=parsed_csv.df,
+            errors_to_return=parsed_csv.errors_to_return,
+            csv_file_name=csv_file_name,
+            submission=submission
+        )
+
+        async_to_sync(tidy_up_old_submissions)(
+            pdu=pdu,
+            new_submission=submission
+        )
+
+        return errors
+    
+    def upload_as_questionnaire_entries(self, audit_year, pdu_pz_code, user, parsed_csv):
+        df = parsed_csv.df
+
+        pz_codes_that_need_submissions = set()
+        pz_codes_that_already_have_submissions = set()
+
+        for pz_code in df["PDU Number"].unique():
+            try:
+                pdu = PaediatricDiabetesUnit.objects.get(pz_code=pz_code)
+            except PaediatricDiabetesUnit.DoesNotExist:
+                raise ValueError(f"Invalid PDU Number: {pz_code}")
```

Comment:
I got this because the source data still referred to PZ080 and PZ141 not PZ250 (pre https://github.com/rcpch/national-paediatric-diabetes-audit/issues/852).

I've manually adjusted the data to reference the correct PDU.